test/rspamd_cxx_unit_cryptobox.hxx

```diff
@@ -0,0 +1,182 @@
+		memset(out2, 0, rspamd_cryptobox_HASHSTATEBYTES);
+
+		rspamd_cryptobox_hash(out2, data, len, key, keylen);
+		CHECK(*out1 == *out2);
+	}
+
+	TEST_CASE("rspamd_cryptobox_fast_hash")
+	{
+		rspamd_cryptobox_fast_hash_state_s *st = rspamd_cryptobox_fast_hash_new();
+
+		uint64_t seed = 10;
+
+		rspamd_cryptobox_fast_hash_init(st, seed);
+
+		const unsigned char* data = reinterpret_cast<const unsigned char *>("data");
+		gsize len = sizeof(data);
+
+		rspamd_cryptobox_fast_hash_update(st, data, len);
+
+		uint64_t out1 = rspamd_cryptobox_fast_hash_final(st);
+
+		uint64_t out2 = rspamd_cryptobox_fast_hash(data, len, seed);
+		CHECK(out1 == out2);
+
+		rspamd_cryptobox_fast_hash_free(st);
+	}
+
+	TEST_CASE("rspamd_cryptobox_pbkdf")
+	{
+		const char *pass = "passpa";
+		gsize pass_len = sizeof(pass);
+
+		const uint8_t salt = 100;
+		gsize salt_len = sizeof(salt);
+
+		uint8_t key = 20;
+		gsize key_len = sizeof(key);
+
+		unsigned int complexity = 10;
+		enum rspamd_cryptobox_pbkdf_type type = RSPAMD_CRYPTOBOX_PBKDF2;
+
+
+		CHECK(rspamd_cryptobox_pbkdf(pass, pass_len, &salt, salt_len, &key, key_len, complexity, type) == true);
+		CHECK(key == 60);
+
+		type = RSPAMD_CRYPTOBOX_CATENA;
+		CHECK(rspamd_cryptobox_pbkdf(pass, pass_len, &salt, salt_len, &key, key_len, complexity, type) == true);
+		CHECK(key == 204);
+	}
+
+/*
+	TEST_CASE("rspamd_cryptobox_encrypt_inplace")
+	{
+		unsigned char* data = (unsigned char *) "data";
+		gsize len = 5;
+		rspamd_nonce_t nonce;
+		rspamd_pk_t pk;
+		rspamd_sk_t sk;
+		rspamd_mac_t sig;
+		enum rspamd_cryptobox_mode mode = RSPAMD_CRYPTOBOX_MODE_NIST;
+
+		ottery_rand_bytes(nonce, sizeof(nonce));
+		ottery_rand_bytes(pk, sizeof(pk));
+		ottery_rand_bytes(sk, sizeof(sk));
+		memset(sig, 0, sizeof(sig));
+
+		rspamd_cryptobox_encrypt_inplace(data, len, nonce, pk, sk, sig, mode);
+		MESSAGE(sig);
+	}
+
+	TEST_CASE("rspamd_cryptobox_decrypt_inplace")
+	{
+		unsigned char* data = (unsigned char *) "data";
+		gsize len = 5;
+		rspamd_nonce_t nonce;
+		rspamd_pk_t pk;
+		rspamd_sk_t sk;
+		rspamd_mac_t sig;
+		enum rspamd_cryptobox_mode mode = RSPAMD_CRYPTOBOX_MODE_NIST;
+
+		ottery_rand_bytes(nonce, sizeof(nonce));
+		ottery_rand_bytes(pk, sizeof(pk));
+		ottery_rand_bytes(sk, sizeof(sk));
+		memset(sig, 0, sizeof(sig));
+
+		CHECK(rspamd_cryptobox_decrypt_inplace(data, len, nonce, pk, sk, sig, mode) == true)
+	}
+
+*/
+	TEST_CASE("rspamd_cryptobox_sign")
+	{
+		enum rspamd_cryptobox_mode mode = RSPAMD_CRYPTOBOX_MODE_NIST;
+		rspamd_sk_t sk;
+		rspamd_pk_t pk;
+		unsigned char * sig = (unsigned char *) "sig";
+		unsigned long long siglen = sizeof(sig);
+		const unsigned char m[] = "data to be signed";
+		size_t mlen = strlen((const char*)m);
+
+		ottery_rand_bytes(sk, sizeof(sk));
+		ottery_rand_bytes(pk, sizeof(pk));
+
+		rspamd_cryptobox_sign(sig, &siglen, m, mlen, sk, mode);
+		bool check_result = rspamd_cryptobox_verify_compat(sig, reinterpret_cast<gsize>(&siglen), m, mlen, pk, sk, mode);
```

Comment:
I'm sorry but I don't understand what you are doing here. Why do we need a **secret** key for verification??